crates/uv-macros/src/lib.rs

```diff
@@ -49,3 +49,56 @@ fn impl_combine(ast: &DeriveInput) -> TokenStream {
     };
     gen.into()
 }
+
+fn get_doc_comment(attr: &Attribute) -> Option<String> {
+    if attr.path().is_ident("doc") {
+        if let syn::Meta::NameValue(meta) = &attr.meta {
+            if let syn::Expr::Lit(expr) = &meta.value {
+                if let syn::Lit::Str(str) = &expr.lit {
+                    return Some(str.value().trim().to_string());
+                }
+            }
+        }
+    }
+    None
+}
+
+#[proc_macro_attribute]
+pub fn collect_constants(_attr: TokenStream, input: TokenStream) -> TokenStream {
+    let ast = parse_macro_input!(input as ItemImpl);
+
+    let constants: Vec<_> = ast
+        .items
+        .iter()
+        .filter_map(|item| {
+            match item {
+                ImplItem::Const(item) => {
+                    let name = item.ident.to_string();
+                    let doc = item.attrs.iter().find_map(get_doc_comment).expect("Missing doc comment");
+                    Some((name, doc))
+                }
+                _ => None,
```

Comment:
Another cool idea is to handle the functions as well with return env var patterns.

```suggestion
            ImplItem::Fn(item)
                if !item.attrs.iter().any(|attr| attr.path().is_ident("attr_hidden")) =>
	            {
                // Extract the environment variable patterns
                if let Some(pattern) = get_env_var_pattern_from_attr(&item.attrs) {
                    let doc = item
                        .attrs
                        .iter()
                        .find_map(get_doc_comment)
                        .unwrap_or_else(|| "No documentation provided.".to_string());
                    Some((pattern, doc))
                } else {
                    None // Skip if pattern extraction fails
                }
            }
            _ => None,
```

Using this extractor

```rust
fn get_env_var_pattern_from_attr(attrs: &[Attribute]) -> Option<String> {
    for attr in attrs {
        if attr.path().is_ident("attr_env_var_pattern") {
            if let Ok(lit_str) = attr.parse_args::<LitStr>() {
                return Some(lit_str.value());
            }
        }
    }
    None
}

#[proc_macro_attribute]
pub fn attr_env_var_pattern(_attr: TokenStream, item: TokenStream) -> TokenStream {
    item
}
```

That way, these will also now be in the docs

```rust
    #[attr_env_var_pattern("UV_INDEX_{name}_USERNAME")]
    pub fn index_username(name: &str) -> String {
        format!("UV_INDEX_{name}_USERNAME")
    }

    /// Generates the environment variable key for the HTTP Basic authentication password.
    #[attr_env_var_pattern("UV_INDEX_{name}_PASSWORD")]
    pub fn index_password(name: &str) -> String {
        format!("UV_INDEX_{name}_PASSWORD")
    }
```